### PR TITLE
Group user attribute by scopes

### PR DESCRIPTION
**Pull request by mpmadhavig**
### Related Issues
- https://github.com/wso2-enterprise/asgardeo-product/issues/12301

### Checklist
- [x] e2e cypress tests locally verified.
- [x] Manual test round performed and verified.
- [x] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### Related PRs
- https://github.com/wso2/identity-apps/pull/3497
- https://github.com/wso2/identity-apps/pull/3521
- https://github.com/wso2/identity-apps/pull/3528/
- https://github.com/wso2/identity-apps/pull/3539

### CODE REVIEW

File: apps/console/src/features/applications/components/settings/attribute-management/attribute-selection-oidc.tsx [hunk 1]
```diff
@@ -0,0 +1,943 @@
...
+                    tempFilterSelectedExternalClaims.push(availableExternalClaim);
+                }
+            }
+        });
+
+        setSelectedExternalClaims(tempFilterSelectedExternalClaims);
+        setFilterSelectedExternalClaims(tempFilterSelectedExternalClaims);
+    }, [ claimConfigurations ]);
+
+    // Stops the UI loader when the component is initialized and have fetched the availableExternalClaims.
```
Comment (brionmario) [inline apps/console/src/features/applications/components/settings/attribute-management/attribute-selection-oidc.tsx:166]: Make this a doc comment.

File: apps/console/src/features/applications/components/settings/attribute-management/attribute-selection-oidc.tsx [hunk 2]
```diff
@@ -0,0 +1,943 @@
...
+        });
+
+        setFilterSelectedExternalClaims(tempFilterSelectedExternalClaims);
+        setSelectedExternalClaims(initialSelectedClaims);
+        setAvailableExternalClaims(initialAvailableClaims);
+        setExternalClaims(initialAvailableClaims.concat(initialSelectedClaims));
+        setInitializationFinished(true);
+    };
+
+    useEffect(() => {
```
Comment (brionmario) [inline apps/console/src/features/applications/components/settings/attribute-management/attribute-selection-oidc.tsx:427]: Use effect should occur before functions.

File: apps/console/src/features/applications/components/settings/attribute-management/attribute-settings.tsx [hunk 3]
```diff
@@ -191,6 +191,11 @@ export const AttributeSettings: FunctionComponent<AttributeSelectionPropsInterfa
     // Manage selected dialect
```
Comment (brionmario) [inline apps/console/src/features/applications/components/settings/attribute-management/attribute-settings.tsx:191]: IMO, these comments are not needed. The var name of the state should infer.

File: apps/console/src/features/applications/components/settings/attribute-management/attribute-settings.tsx [hunk 4]
```diff
@@ -221,6 +226,165 @@ export const AttributeSettings: FunctionComponent<AttributeSelectionPropsInterfa
 
     const eventPublisher: EventPublisher = EventPublisher.getInstance();
 
+    /**
+     * Retrieves scopes with the details from the API.
+     */
+    const getScopes = () => {
+
+        getOIDCScopesList()
```
Comment (brionmario) [inline apps/console/src/features/applications/components/settings/attribute-management/attribute-settings.tsx:234]: Use this https://github.com/wso2/identity-apps/blob/master/apps/console/src/features/oidc-scopes/api/oidc-scopes.ts#L83 instead.

Refer: https://github.com/wso2/identity-apps/blob/f9dde022fc4b18a19a503e5d1ec223926daf1904/apps/console/src/features/oidc-scopes/pages/oidc-scopes.tsx#L84

File: apps/console/src/features/applications/components/settings/attribute-management/attribute-settings.tsx [hunk 5]
```diff
@@ -798,11 +1015,16 @@ export const AttributeSettings: FunctionComponent<AttributeSelectionPropsInterfa
     };
 
     useEffect(() => {
```
Comment (brionmario) [inline apps/console/src/features/applications/components/settings/attribute-management/attribute-settings.tsx:1017]: Move useEffects above all the functions.

File: modules/theme/src/themes/default/collections/table.overrides [hunk 6]
```diff
@@ -27,6 +27,13 @@
     }
 }
 
+.ui.table {
```
Comment (brionmario) [inline modules/theme/src/themes/default/collections/table.overrides:30]: Is this style reusable in other portals.
IMO, no. And this should come in `developer-portal.less`(Style sheet meant for Console).

File: apps/console/src/features/applications/components/settings/attribute-management/attribute-selection-oidc.tsx (hunk 1 above)
Comment (mpmadhavig) [reply apps/console/src/features/applications/components/settings/attribute-management/attribute-selection-oidc.tsx]: resolved with https://github.com/wso2/identity-apps/pull/3566/commits/66c501e3bafdfeba484d242847e688d8d0ee2611

File: apps/console/src/features/applications/components/settings/attribute-management/attribute-settings.tsx (hunk 3 above)
Comment (mpmadhavig) [reply apps/console/src/features/applications/components/settings/attribute-management/attribute-settings.tsx]: resolved with https://github.com/wso2/identity-apps/pull/3566/commits/66c501e3bafdfeba484d242847e688d8d0ee2611

File: apps/console/src/features/applications/components/settings/attribute-management/attribute-settings.tsx (hunk 4 above)
Comment (mpmadhavig) [reply apps/console/src/features/applications/components/settings/attribute-management/attribute-settings.tsx]: resolved with https://github.com/wso2/identity-apps/pull/3566/commits/66c501e3bafdfeba484d242847e688d8d0ee2611

File: modules/theme/src/themes/default/collections/table.overrides (hunk 6 above)
Comment (mpmadhavig) [reply modules/theme/src/themes/default/collections/table.overrides:30]: resolved with https://github.com/wso2/identity-apps/pull/3566/commits/66c501e3bafdfeba484d242847e688d8d0ee2611

File: apps/console/src/features/applications/components/settings/attribute-management/attribute-settings.tsx (hunk 5 above)
Comment (mpmadhavig) [reply apps/console/src/features/applications/components/settings/attribute-management/attribute-settings.tsx]: resolved with https://github.com/wso2/identity-apps/pull/3566/commits/66c501e3bafdfeba484d242847e688d8d0ee2611

File: apps/console/src/features/applications/components/settings/attribute-management/attribute-selection-oidc.tsx (hunk 2 above)
Comment (mpmadhavig) [reply apps/console/src/features/applications/components/settings/attribute-management/attribute-selection-oidc.tsx]: resolved with https://github.com/wso2/identity-apps/pull/3566/commits/66c501e3bafdfeba484d242847e688d8d0ee2611

File: apps/console/src/features/applications/components/settings/attribute-management/attribute-settings.tsx [hunk 7]
```diff
@@ -28,6 +28,9 @@ import {
 } from "@wso2is/core/models";
 import { addAlert } from "@wso2is/core/store";
 import { ConfirmationModal, ContentLoader, EmphasizedSegment } from "@wso2is/react-components";
+import { useOIDCScopesList } from "apps/console/src/features/oidc-scopes/api/oidc-scopes";
+import { OIDCScopesClaimsListInterface, OIDCScopesListInterface
```
Comment (brionmario) [inline apps/console/src/features/applications/components/settings/attribute-management/attribute-settings.tsx:32]:
```suggestion
import {
    OIDCScopesClaimsListInterface,
    OIDCScopesListInterface 
```

File: modules/theme/src/themes/default/collections/table.overrides (hunk 6 above)
Comment (brionmario) [reply modules/theme/src/themes/default/collections/table.overrides:30]: Lets move `.thead.user-attribute-table-header {` as well.

**Review by brionmario — DISMISSED** [review]

File: apps/console/src/features/applications/components/settings/attribute-management/attribute-list-item.tsx [hunk 8]
```diff
@@ -212,41 +258,33 @@ export const AttributeListItem: FunctionComponent<AttributeListItemPropInterface
...
+                        <Checkbox
+                            checked={ initialRequested || requested || subject }
+                            onClick={ !readOnly && handleRequestedCheckChange }
+                            disabled={ mappingOn ? !mandatory : false }
+                            readOnly={ subject || readOnly || isOIDCMapping }
+                        />
+                    </Table.Cell>
+                )
+            }
+
```
Comment (brionmario) [inline apps/console/src/features/applications/components/settings/attribute-management/attribute-list-item.tsx:277]:
```suggestion
```

**Review by brionmario — APPROVED** [review]

**Pull request merged**